test/support/index.js

```diff
@@ -1,86 +1,23 @@
-import DashboardPage from './pages/dashboard_page';
-import ShieldPage from './pages/shield_page';
-import ConsolePage from './pages/console_page';
 
+// Intern values provided via the root index file of the test suite.
 const kbnInternVars = global.__kibana__intern__;
-
 exports.intern = kbnInternVars.intern;
-exports.config = exports.intern.config;
-exports.defaultTimeout = exports.config.defaultTimeout;
-exports.defaultTryTimeout = exports.config.defaultTryTimeout;
-exports.defaultFindTimeout = exports.config.defaultFindTimeout;
-exports.scenarioManager = new ScenarioManager(url.format(exports.config.servers.elasticsearch));
-exports.esClient = new EsClient(url.format(exports.config.servers.elasticsearch));
 exports.bdd = new BddWrapper(kbnInternVars.bdd);
 
-const delayedExports = [{
-  name: 'remote',
-  exportProvider: (suite) => suite.remote
-}, {
-  name: 'common',
-  exportProvider: () => new Common()
-}, {
-  name: 'discoverPage',
-  exportProvider: () => new DiscoverPage()
-}, {
-  name: 'headerPage',
-  exportProvider: () => new HeaderPage()
-}, {
-  name: 'settingsPage',
-  exportProvider: () => new SettingsPage()
-}, {
-  name: 'visualizePage',
-  exportProvider: () => new VisualizePage()
-}, {
-  name: 'dashboardPage',
-  exportProvider: () => new DashboardPage()
-}, {
-  name: 'shieldPage',
-  exportProvider: () => new ShieldPage()
-}, {
-  name: 'consolePage',
-  exportProvider: () => new ConsolePage()
-}, {
-  name: 'elasticDump',
-  exportProvider: () => new ElasticDump()
-}];
-
-delayedExports.forEach(exportConfig => {
-  delayExportUntilInit(
-    exportConfig.name,
-    exportConfig.exportProvider
-  );
-});
-
-// Creates an export for values that aren't actually avaialable until
-// tests start to run. These getters will throw errors if the export
-// is accessed before it's available, hopefully making debugging easier.
-// Once the first before() handler is called the onInit() call
-// will fire and rewrite all of these exports to be their correct value.
-function delayExportUntilInit(name, provideExport) {
-  Object.defineProperty(exports, name, {
-    configurable: true,
-    get() {
-      throw new TypeError(
-        'Remote is not available until tests start to run. Move your ' +
-        'usage of the import inside a test setup hook or a test itself.'
-      );
-    }
-  });
+// Config options
+const config = exports.config = kbnInternVars.intern.config;
+exports.defaultTimeout = config.defaultTimeout;
+exports.defaultTryTimeout = config.defaultTryTimeout;
+exports.defaultFindTimeout = config.defaultFindTimeout;
 
-  kbnInternVars.onInit(function defineExport() {
-    Object.defineProperty(exports, name, {
-      value: provideExport(this),
-    });
-  });
-}
+// Helper instances
+exports.scenarioManager =
+  new ScenarioManager(url.format(config.servers.elasticsearch));
+exports.elasticDump = new ElasticDump();
+exports.esClient = new EsClient(url.format(config.servers.elasticsearch));
```

Comment:
Maybe we can extract these from this file, and handle configuration and initialization somewhere else (e.g. inside of the functional tests index.js), which would remove the need for this file entirely.